packages/concerto-core/lib/introspect/decorator.js

```diff
@@ -83,7 +84,10 @@ class Decorator {
                             name: thing.type.name,
                             array: thing.isArray
                         });
-                    } else {
+                    } else if(thing.$class.includes('DecoratorTypeReference')){
```

Comment:
I'm skeptical if this resolves the issue.

Have gone through the links I had shared in the last comment?
Can you please describe what is your understanding/interpretation of the problem and how are you trying to solve it?
I feel there are some gaps in interpretation of the task in hand, I can help fill those.

Reply:
What I was trying to do is throw an error on arguments that have a DecoratorTypeReference but are not in the given namespace. 

I've gone now to links you mentioned and It seems to me that I mixed up between "DecoratorTypeReference" and "type reference argument."

What I should be doing is resolving arguments that are neither boolean, nor string, nor numbers and throwing an error if it has a different namespace.

Is my interpretation correct?

Reply:
Yes exactly! So for this we have someone contributing to build a way to resolve if the decorator type reference is present in the namespace or now.
https://github.com/accordproject/concerto/issues/797
You might want to discuss with them as well what approach are they taking.

Reply:
ok, thanks I'll do so.

and for testing I should add this car file in **(packages/concerto-core/test/data/model/)** Directory and throw error test in **(packages/concerto-core/test/modelmanager.js)** is that right?

Reply:
It depends where you implement the validation(resolving decorator type reference ) logic. If you are are planning to add the logic in `decorator.js` then your test should also reside in file that does test for the class and methods present in `decorator.js`